### ISSUE TITLE
Content warning toggle buttons are missing ARIA roles

### ISSUE THREAD

**Issue by zichy** [mastodon/mastodon#19920]
### Steps to reproduce the problem

1. Navigate to any CW toggle button (`.status__content__spoiler-link`) using assistive technology such as a screen reader 
2. The button does not announce its state in a machine-readable way

### Expected behaviour

The screen reader should announce the button's state without relying fully on its text

### Actual behaviour

The screen reader speaks "show more/less" but is missing the standardized and expected hint regarding its toggling fuction

### Detailed description

The button should get the [`aria-expanded`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) attribute and change the value to `true` or `false` depending on its state. This way, the screen reader will read "collapsed" or "expanded".

The button could refer to the hidden content by using `aria-controls`, but the [screen reader support](https://a11ysupport.io/tech/aria/aria-controls_attribute) is pretty bad.

The button does not need a `tabindex` attribute since it can natively receive keyboard focus.

### Specifications

Mastodon 4.0.0rc1
Safari 16.1
macOS 13.0
VoiceOver

**Comment by ClearlyClaire** [comment]
Should be addressed by #19975 but your feedback would be greatly appreciated. I also wonder if we should do something more explicit regarding how the post's contents are hidden when the CW is collapsed.

**Comment by zichy** [comment]
@ClearlyClaire Thank you, that was quick! If there's no technical reason against it, using `aria-controls` wouldn't hurt.

Regarding the general toggling mechanic: I'd generally prefer using `<details>`, which could be styled in a way that wouldn't be different from the current implementation. Then I saw that mentioned names in toots aren't actually hidden, which makes it more complicated. I don't think that anything "more explicit" is needed regarding how the content is hidden.
